scripts/ingest-tarball.sh

```diff
@@ -193,11 +193,15 @@ function ingest_compat_tarball() {
     if [ -f "/cvmfs/${repo}/${basedir}/${version}/compat/${os}/${arch}/startprefix" ];
     then
         echo_yellow "Compatibility layer for version ${version}, OS ${os}, and architecture ${arch} already exists!"
-        echo_yellow "Removing the existing layer, and adding the new one from the tarball..."
+        old_layer_path="/cvmfs/${repo}/${basedir}/${version}/compat/${os}/${arch}"
```

Comment:
I would change this a little bit since we're using the same path above (untested):

```shell
    compat_layer_path="/cvmfs/${repo}/${basedir}/${version}/compat/${os}/${arch}"
    if [ -f "${compat_layer_path}/startprefix" ];
    then
        echo_yellow "Compatibility layer for version ${version}, OS ${os}, and architecture ${arch} already exists!"
        echo_yellow "Removing the existing layer, and adding the new one from the tarball..."

        ${cvmfs_server} transaction "${repo}"

        last_suffix=$((ls -1d ${compat_layer_path}-* | tail -n 1 | xargs basename | cut -d- -f2) 2> /dev/null)
        new_suffix=$(printf '%03d\n' $((${last_suffix:-0} + 1)))
        old_layer_suffixed_path="${compat_layer_path}-${new_suffix}"
        echo_yellow "Moving the existing compat layer from ${compat_layer_path} to ${old_layer_suffixed_path}..."
        mv ${compat_layer_path} ${old_layer_suffixed_path}
        ...

Reply:
thanks, done in https://github.com/EESSI/filesystem-layer/pull/176/commits/bc2a3aaecddbc5533108a44933c4b2989ef05d49